client/packages/common/src/ui/layout/tables/components/Cells/NumberInputCell/NonNegativeDecimalCell.tsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { CellProps } from '../../../columns';
+import { NonNegativeNumberInput } from '@common/components';
+import { RecordWithId } from '@common/types';
+import { useBufferState, useDebounceCallback } from '@common/hooks';
+
+// where NonNegative is n >=0
+export const NonNegativeDecimalCell = <T extends RecordWithId>({
+  rowData,
+  column,
+  rows,
+  isDisabled = false,
+}: CellProps<T>): React.ReactElement<CellProps<T>> => {
+  const [buffer, setBuffer] = useBufferState(
+    column.accessor({ rowData, rows })
+  );
+
+  const updater = useDebounceCallback(column.setter, [column.setter], 250);
+
+  return (
+    <NonNegativeNumberInput
+      disabled={isDisabled}
+      InputProps={{ sx: { '& .MuiInput-input': { textAlign: 'right' } } }}
+      type="number"
+      value={buffer}
```

Comment:
i haven't tested this, but a question, if value is already a bit decimal, would it be truncated ?

Reply:
The input is to 2 decimal places, but if an inbound invoice has been pulled from central and the number is higher than 2 dp then it will show that decimal instead, should I truncate it to 2 dp? Though that would void [Mark's comment](https://github.com/openmsupply/open-msupply/pull/647#pullrequestreview-1121111624)... 

Other option is just to change the input back to a whole number for inbound shipments

Reply:
Let leave, can discuss in our next meeting, thanks